v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/ElasticsearchTransforms.java

```diff
@@ -0,0 +1,326 @@
+ * the License.
+ */
+package com.google.cloud.teleport.v2.transforms;
+
+import static org.apache.beam.vendor.guava.v20_0.com.google.common.base.Preconditions.checkArgument;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.auto.value.AutoValue;
+import com.google.cloud.teleport.v2.transforms.JavascriptTextTransformer.JavascriptRuntime;
+import java.io.IOException;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.script.ScriptException;
+import org.apache.beam.sdk.io.elasticsearch.ElasticsearchIO;
+import org.apache.beam.sdk.io.elasticsearch.ElasticsearchIO.ConnectionConfiguration;
+import org.apache.beam.sdk.io.elasticsearch.ElasticsearchIO.RetryConfiguration;
+import org.apache.beam.sdk.io.elasticsearch.ElasticsearchIO.Write.FieldValueExtractFn;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.Validation.Required;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.joda.time.Duration;
+
+/** Contains common transforms for Elasticsearch such as reading and writing. */
+public class ElasticsearchTransforms {
+
+  /** Convert provided long to {@link Duration}. */
+  private static Duration getDuration(Long milliseconds) {
+    return new Duration(milliseconds);
+  }
+
+  /**
+   * The {@link WriteToElasticsearchOptions} class provides the custom execution options passed by
+   * the executor at the command-line.
+   */
+  public interface WriteToElasticsearchOptions extends PipelineOptions {
+    @Description(
+        "Comma separated list of Elasticsearch nodes to connect to, ex: http://my-node1,http://my-node2")
+    @Required
+    String getNodeAddresses();
+
+    void setNodeAddresses(String nodeAddresses);
+
+    @Description("The index toward which the requests will be issued, ex: my-index")
+    @Required
+    String getIndex();
+
+    void setIndex(String index);
+
+    @Description("The document type toward which the requests will be issued, ex: my-document-type")
+    @Required
+    String getDocumentType();
+
+    void setDocumentType(String documentType);
+
+    @Description("Batch size in number of documents. Default: 1000")
+    @Default.Long(1000)
+    Long getBatchSize();
+
+    void setBatchSize(Long batchSize);
+
+    @Description("Batch size in number of bytes. Default: 5242880 (5mb)")
+    @Default.Long(5242880)
+    Long getBatchSizeBytes();
+
+    void setBatchSizeBytes(Long batchSizeBytes);
+
+    @Description("Optional: Max retry attempts, must be > 0, ex: 3. Default: no retries")
+    Integer getMaxRetryAttempts();
+
+    void setMaxRetryAttempts(Integer maxRetryAttempts);
+
+    @Description(
+        "Optional: Max retry duration in milliseconds, must be > 0, ex: 5000L. Default: no retries")
```

Comment:
We should enforce this validation in the build()

Reply:
Yes definitely. Both of the retry config parameters are validated in build().